### PR TITLE
Delete partition request

### PR DESCRIPTION
**Pull request by idruzhitskiy**

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -109,6 +109,25 @@ paths:
                 type: string
         default:
           $ref: '#/components/responses/DefaultError'
+  /vdisks/{vdiskId}/partitions/{partitionId}:
```
Comment (pyakushin) [inline openapi.yaml]: {VDiskId}, you can check at editor.swagger.io

File: openapi.yaml [hunk 2]
```diff
@@ -109,6 +109,25 @@ paths:
...
+    delete:
```
Comment (pyakushin) [inline openapi.yaml]: API already has that path, add just `delete` method to it

File: src/api/http.rs [hunk 3]
```diff
@@ -234,7 +235,7 @@ fn change_partition_state(
     let task = async move {
         match action {
             Action::Attach => group.attach(partition_id).await,
-            Action::Detach => group.detach(partition_id).await,
+            Action::Detach => group.detach(partition_id).await.map(|_| ()),
```
Comment (pyakushin) [inline src/api/http.rs]: Why do you need to map to zero type?

Comment (pyakushin) [reply src/api/http.rs]: I get it, maybe just:
```
{
    group.detach(partition_id).await;
}
```
?

File: src/core_inner/backend/pearl/group.rs [hunk 4]
```diff
@@ -308,8 +308,9 @@ impl PearlGroup {
                         warn!("pearl closed: {:?}", e);
                     }
                 }
+                let pearl = pearl.clone();
```
Comment (pyakushin) [inline src/core_inner/backend/pearl/group.rs]: Is it possible to find pearl and remove it, to prevent clone?

Comment (idruzhitskiy) [reply src/core_inner/backend/pearl/group.rs]: I'll check

File: src/api/http.rs (hunk 3 above)
Comment (idruzhitskiy) [reply src/api/http.rs]: Oh, sure, missed that opportunity :)

Comment (idruzhitskiy) [reply src/api/http.rs]: Oh, wait, actually we need error from that result, and map applies to Ok only

File: src/core_inner/backend/pearl/group.rs (hunk 4 above)
Comment (idruzhitskiy) [reply src/core_inner/backend/pearl/group.rs]: Done

**Comment by pyakushin** [comment]
My tests
- [ ] deny to delete active partition
- [ ] inactive partition delete
- [ ] inactive partition delete under the load

**Pull request merged**